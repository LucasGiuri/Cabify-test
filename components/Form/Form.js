import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import './Form.scss';
import Selectors from '../Selectors/Selectors';
import Button from '../Button/Button';

const Form = ({selectors, total, onAddClick, onRemoveClick, onSelectChange}) => {
  return (
    <Fragment>
      <Selectors selectors={selectors} onChange={onSelectChange} onClick={onRemoveClick} />
      <div className="form--add-button">
        <Button text="Add" onClick={onAddClick} />
      </div>
      <h4>Total: {total}</h4>
    </Fragment>
  )
}

Form.propTypes = {
  selectors: PropTypes.arrayOf(PropTypes.object).isRequired,
  total: PropTypes.number.isRequired,
  onAddClick: PropTypes.func,
  onRemoveClick: PropTypes.func,
  onSelectChange: PropTypes.func
};

Form.defaultProps = {
  onAddClick: () => '',
  onRemoveClick: () => '',
  onSelectChange: () => ''
};

export default Form;