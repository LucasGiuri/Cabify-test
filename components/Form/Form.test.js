import React from 'react';
import renderer from 'react-test-renderer';
import Form from './Form';

jest.mock('../Selectors/Selectors', () => 'Selectors');
jest.mock('../Button/Button', () => 'Button');

describe('Form', () => {
  it('Form basic render snapshoot', () => {
    const selectors = [{}];
    const component = renderer.create(
      <Form selectors={selectors} total={1} onAddClick={jest.fn()} onSelectChange={jest.fn()} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })


  it('Form prop onAddClick is triggered. onClick prop should be called as expected', () => {
    const onAddClick = jest.fn();
    const selectors = [{}];
    const componentToTest = (
      <Form selectors={selectors} total={1} onAddClick={onAddClick} onSelectChange={jest.fn()} />
    );
    const tree = renderer.create(componentToTest);
    const button = tree.root.findByType("Button");
    button.props.onClick();
    expect(onAddClick).toHaveBeenCalledTimes(1);
    expect(onAddClick).toHaveBeenCalledWith();
  })
})
