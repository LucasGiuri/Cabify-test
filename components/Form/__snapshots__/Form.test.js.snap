// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Form basic render snapshoot 1`] = `
Array [
  <Selectors
    onChange={[MockFunction]}
    onClick={[Function]}
    selectors={
      Array [
        Object {},
      ]
    }
  />,
  <div
    className="form--add-button"
  >
    <Button
      onClick={[MockFunction]}
      text="Add"
    />
  </div>,
  <h4>
    Total: 
    1
  </h4>,
]
`;
